app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -405,21 +593,406 @@ private fun AssociationMembers(
+  if (showEditDialog) {
+    SaveRoleDialog(
+        onDismiss = { showEditDialog = false },
+        onCreateRole = { updatedRole ->
+          associationViewModel.selectedAssociation.value?.let { association ->
+            associationViewModel.editRoleLocally(association.uid, updatedRole)
+          }
+          showEditDialog = false
+        },
+        associationViewModel = associationViewModel,
+        initialRole = role)
+  }
+
+  // Delete Role Confirmation Dialog
+  if (showDeleteDialog) {
+    AlertDialog(
+        onDismissRequest = { showDeleteDialog = false },
+        confirmButton = {
+          Button(
+              onClick = {
+                associationViewModel.selectedAssociation.value?.let { association ->
+                  associationViewModel.deleteRoleLocally(association.uid, role)
+                }
+                showDeleteDialog = false
+              }) {
+                Text(context.getString(R.string.association_profile_role_card_delete))
+              }
+        },
+        dismissButton = {
+          TextButton(onClick = { showDeleteDialog = false }) {
+            Text(context.getString(R.string.association_profile_role_card_cancel))
+          }
+        },
+        title = { Text(context.getString(R.string.association_profile_role_card_delete_role)) },
+        text = {
+          Text(
+              context.getString(R.string.association_profile_role_card_sure_delete_role) +
+                  " '${role.displayName}'?")
+        })
+  }
+}
+
+/**
+ * Displays a dialog to create or edit a role within the association. If an existing role is passed,
+ * the dialog will allow editing the role's details, including the display name, color, and
+ * permissions. If no existing role is passed, the dialog will create a new role. Once the role is
+ * saved, it triggers the appropriate actions in the [associationViewModel].
+ *
+ * @param onDismiss A lambda function to be called when the dialog is dismissed.
+ * @param onCreateRole A lambda function to be called after the role is created or updated, passing
+ *   the role.
+ * @param associationViewModel The ViewModel responsible for managing the association's data.
+ * @param initialRole The initial role details to prefill the dialog fields (optional, null if
+ *   creating a new role).
+ */
+@Composable
+fun SaveRoleDialog(
+    onDismiss: () -> Unit,
+    onCreateRole: (Role) -> Unit,
+    associationViewModel: AssociationViewModel,
+    initialRole: Role? = null
+) {
+  var displayName by remember { mutableStateOf(TextFieldValue(initialRole?.displayName ?: "")) }
+  val controller = rememberColorPickerController()
+  var selectedColor by remember {
+    mutableStateOf(Color(initialRole?.color ?: Color.White.toArgb().toLong()))
+  }
+  val selectedPermissions = remember {
+    mutableStateListOf<PermissionType>().apply {
+      initialRole?.permissions?.getGrantedPermissions()?.let { addAll(it) }
+    }
+  }
+  val allPermissions = PermissionType.values()
```

Comment:
Doesn't seem super important, but my ide suggest to switch for .entries() as .values is deprecated